src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php

```diff
@@ -0,0 +1,354 @@
+     */
+    public function iDeleteTheProductAssociationType(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        $this->client->delete($productAssociationType->getCode());
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     * @Then I should be notified that they have been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted(): void
+    {
+        Assert::true($this->responseChecker->isDeletionSuccessful(
+            $this->client->getLastResponse()),
+            'Product association type could not be deleted'
+        );
+    }
+
+    /**
+     * @Then /^(this product association type) should no longer exist in the registry$/
+     */
+    public function thisProductAssociationTypeShouldNoLongerExistInTheRegistry(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($this->client->index(), 'code', $productAssociationType->getCode()),
+            sprintf('Product association type with code %s exist', $productAssociationType->getCode())
+        );
+    }
+
+    /**
+     * @When I want to modify the :productAssociationType product association type
+     */
+    public function iWantToModifyTheProductAssociationType(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        $this->client->buildUpdateRequest($productAssociationType->getCode());
+    }
+
+    /**
+     * @When I rename it to :name in :language
+     */
+    public function iRenameItToIn(string $name, string $language): void
+    {
+        $this->client->updateRequestData(['translations' => [$language => ['name' => $name, 'locale' => $language]]]);
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
+    {
+        Assert::true(
+            $this->responseChecker->isUpdateSuccessful($this->client->getLastResponse()),
+            'Product association type could not be edited'
+        );
+    }
+
+    /**
+     * @Then this product association type name should be :name
+     */
+    public function thisProductAssociationTypeNameShouldBe(string $name): void
+    {
+        /** @var ProductAssociationType $productAssociationType */
+        $productAssociationType = $this->sharedStorage->get('product_association_type');
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->show($productAssociationType->getCode()), 'name', $name),
+            sprintf('Product association type name is not %s', $name)
+        );
+    }
+
+    /**
+     * @Then the code field should be disabled
```

Comment:
During implementing new steps, we have decided, that we would like to slightly adjust these steps to something like https://github.com/Sylius/Sylius/pull/11178/files#diff-32a4b0d263dad520cfb84a0704f82849R24